client/src/main/java/com/mytiki/publish/client/permission/PermissionActivity.kt

```diff
@@ -0,0 +1,178 @@
+package com.mytiki.publish.client.permission
+
+import android.Manifest
+import android.annotation.SuppressLint
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Build
+import android.os.Bundle
+import android.provider.Settings
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import com.mytiki.publish.client.TikiClient
+
+class PermissionActivity : AppCompatActivity() {
+
+  private val callbackMap = mutableMapOf<Permission, Boolean>()
+  private var diferentPermissions = mutableListOf<Permission>()
+  private var normalPermissions = mutableListOf<Permission>()
+  private var isNotificationPermissionRequested = false
+  private var trackingPermissionLauncher: ActivityResultLauncher<Intent>? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    val permissions =
+        intent.extras?.getStringArray("permissions")?.mapNotNull { Permission.fromString(it) }
```

Comment:
It should not use intent for this. The onCreate should be clean and use a static method to request the permissions, passing the object.